c++/src/capnp/any.c++

```diff
@@ -77,4 +79,127 @@ kj::Own<ClientHook> AnyPointer::Pipeline::asCap() {
 
 #endif  // !CAPNP_LITE
 
+StructEqualityResult equal(AnyStruct::Reader left, AnyStruct::Reader right) {
+  auto dataL = left.getDataSection();
+  size_t dataSizeL = dataL.size();
+  while(dataSizeL > 0 && dataL[dataSizeL - 1] == 0) {
+    -- dataSizeL;
+  }
+
+  auto dataR = right.getDataSection();
+  size_t dataSizeR = dataR.size();
+  while(dataSizeR > 0 && dataR[dataSizeR - 1] == 0) {
+    -- dataSizeR;
+  }
+
+  if(dataSizeL != dataSizeR) {
+    return StructEqualityResult::NOT_EQUAL;
+  }
+
+  if(0 != memcmp(dataL.begin(), dataR.begin(), dataSizeL * sizeof(word))) {
+    return StructEqualityResult::NOT_EQUAL;
+  }
+
+  auto ptrsL = left.getPointerSection();
+  auto ptrsR = right.getPointerSection();
+
+  size_t i = 0;
+
+  for(; i < kj::min(ptrsL.size(), ptrsR.size()); i++) {
+    auto l = ptrsL[i];
+    auto r = ptrsR[i];
+    switch(equal(l, r)) {
+    case StructEqualityResult::EQUAL:
```

Comment:
I think I mentioned this before, but how do you feel about having an auto-formatter config file checked in?  I'm personally a fan of clang-format.